src/main/java/uk/gov/hmcts/reform/sscs/service/hmc/topic/ProcessHmcMessageService.java

```diff
@@ -26,64 +31,84 @@
-                caseId,
-                hearingId,
-                hmcMessage.getHmctsServiceCode());
+        HmcStatus hmcMessageStatus = hmcMessage.getHearingUpdate().getHmcStatus();
 
         HearingGetResponse hearingResponse = hmcHearingApiService.getHearingRequest(hearingId);
+
         HmcStatus hmcStatus = hearingResponse.getRequestDetails().getStatus();
 
-        if (hmcMessageStatus != hmcStatus) {
-            throw new InvalidHmcMessageException(String.format("HMC Message Status '%s' does not match the GET request status '%s' "
-                    + "for Case ID %s and Hearing ID %s",
-                    hmcMessageStatus, hmcStatus, caseId, hearingId));
+        checkStatuses(caseId, hearingId, hmcMessageStatus, hmcStatus);
+
+        if (stateNotHandled(hmcStatus, hearingResponse)) {
+            log.info("CCD state has not been updated for the Hearing ID {} and Case ID {}",
+                hearingId, caseId);
+            return;
         }
 
         SscsCaseData caseData = ccdCaseService.getCaseDetails(caseId).getData();
 
         if (isHearingUpdated(hmcStatus, hearingResponse)) {
-            caseStateUpdateService.updateListed(hearingResponse, caseData);
-        } else if (isHearingCancelled(hmcStatus, hearingResponse)) {
-            caseStateUpdateService.updateCancelled(hearingResponse, caseData);
-        } else if (isStatusException(hmcStatus)) {
-            caseStateUpdateService.updateFailed(caseData);
-        } else {
-            log.info("CCD state has not been updated for the Hearing ID {} and Case ID {}",
-                    hearingId,
-                    caseId);
-            return;
+            hearingUpdateService.updateHearing(hearingResponse, caseData);
         }
 
         String ccdUpdateDescription = String.format(hmcStatus.getCcdUpdateDescription(), hearingId);
-        ccdCaseService.updateCaseData(caseData,
-                hmcStatus.getCcdUpdateEventType(),
-                hmcStatus.getCcdUpdateSummary(),
-                ccdUpdateDescription);
+
+        resolveEventAndUpdateCase(hearingResponse, hmcStatus, caseData, ccdUpdateDescription);
 
         log.info("Hearing message {} processed for case reference {}",
                 hmcMessage.getHearingId(),
                 hmcMessage.getCaseId());
     }
 
+    private void resolveEventAndUpdateCase(HearingGetResponse hearingResponse, HmcStatus hmcStatus, SscsCaseData caseData,
+                           String ccdUpdateDescription) throws UpdateCaseException {
+
+        Function<HearingGetResponse, EventType> eventMapper = hmcStatus.getEventMapper();
+
+        if (eventMapper != null) {
+            EventType eventType = eventMapper.apply(hearingResponse);
+            if (eventType != null) {
+                ccdCaseService.updateCaseData(caseData,
+                    eventType,
+                    hmcStatus.getCcdUpdateSummary(),
+                    ccdUpdateDescription);
+            }
+        }
+    }
```

Comment:
Minor: Better to get out early.  I.e.
if(eventMapper == null ) {
return;
}
...

This way its easier to read with less nested if's